### ISSUE TITLE
Can't milk a robot, others

### ISSUE THREAD

**Issue by SunshineDistillery** [CleverRaven/Cataclysm-DDA#25519]
**Describe the bug**  
As far as I can see, there are three ways a monster can be friendly: being defined as belonging to the player's faction (slimespring), becoming flagged as a pet (cow), being flagged as controlled (turrets presumably). These three friendlies don't have the same behavior. A trait like milking is available for a tamed creature, but a milkable robot can only be turned off. 

**To Reproduce**  
Feed a cow and release a manhack. "e"xamining them brings up different menus/options. 

**Expected behavior**  
One 'friendly' state and a standard menu for interacting with them.  Monster traits, like milking, to work consistently no matter how the monster was tamed or deployed.

**Versions and configuration:**
 - OS: Windows 7 
 - Game Version: 7619
 - Graphics version: Tiles
 - Mods loaded: None/ testing mods

**Additional context**  
By itself, a pretty inconsequential problem, but there are other things like putting a bag on a robot or tying a robot to a leash that could be useful. Things like the flag PET_WONT_FOLLOW doesn't work on deployed grenade-hacks, which would be really nice.

**Comment by 0be1isk** [comment]
uhhhh what

**Comment by 0be1isk** [comment]
no offense but, I'e read this over 2 times, and besides the friendly units mechanic, this seems very nonsensical

**Comment by kevingranade** [comment]
It's a infrastructure issue that is impacting ability to do modding I think.  The feature ask is that no matter how a creature gets a "tamed" flag, they can be treated the same way with respect to interactions.

**Comment by SunshineDistillery** [comment]
Yeah, exactly.  There are several 'friendly' states, but there should be one.  Currently, useful features for pets don't apply to turrets/robots just because those monsters are spawned from items. 

Obviously, milking a robot is silly, but I thought I could hijack the feature to mod robots to produce mutagen or other chemicals.  From there I discovered a few other issues resulting from having separate flags.
